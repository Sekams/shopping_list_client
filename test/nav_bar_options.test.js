import React from 'react';
import NavBarOptions from '../src/components/nav_bar_options'

test("renders the nav_bar_options component", () => {
    const wrapper = shallow(
        <NavBarOptions />
    );
    expect(wrapper).toMatchSnapshot();
});

test("calls the onSearchTermChanged", () => {
    const props = {
        onSearchTermChange: jest.fn(),
    };
    const wrapper = shallow(
        <NavBarOptions {...props} />
    );
    wrapper.props().children.map((child) => {
        child.props.onSearchTermChange();
    });
    expect(wrapper).toMatchSnapshot();
});