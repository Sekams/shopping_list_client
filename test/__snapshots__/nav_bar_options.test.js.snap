// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calls the onSearchTermChanged 1`] = `
<ul
  className="nav navbar-nav navbar-right"
>
  <NavBarOptionItem
    key="search"
    name="search"
    onSearchTermChange={[Function]}
  />
  <NavBarOptionItem
    key="greeting"
    name="greeting"
    onSearchTermChange={[Function]}
  />
  <NavBarOptionItem
    key="link"
    name="link"
    onSearchTermChange={[Function]}
  />
</ul>
`;

exports[`renders the nav_bar_options component 1`] = `
<ul
  className="nav navbar-nav navbar-right"
>
  <NavBarOptionItem
    key="search"
    name="search"
    onSearchTermChange={[Function]}
  />
  <NavBarOptionItem
    key="greeting"
    name="greeting"
    onSearchTermChange={[Function]}
  />
  <NavBarOptionItem
    key="link"
    name="link"
    onSearchTermChange={[Function]}
  />
</ul>
`;
